test_files/typedef/typedef.ts

```diff
@@ -3,4 +3,15 @@ var y: MyType = 3;
 
 type Recursive = {value: number, next: Recursive};
 
+// TypeScript expands union members when constructing, which drops the
+// distinction in unionUserInUnion between string|number|boolean and
+// UnionTypeAlias|boolean. That's a problem if the constituent types are not
+// valid symbols in the lexical context (e.g. not imported).
```

Comment:
You write the problem here, but trail off before describing why the problem isn't a problem.  Either briefly describe what you expect the test to be showing here, or ref the relevant code or something?

Reply:
done.